src/main/sensors/voltage.c

```diff
@@ -182,8 +199,25 @@ void voltageMeterADCRead(voltageSensorADC_e adcChannel, voltageMeter_t *voltageM
 {
     voltageMeterADCState_t *state = &voltageMeterADCStates[adcChannel];
 
-    voltageMeter->filtered = state->voltageFiltered;
+    voltageMeter->displayFiltered = state->voltageDisplayFiltered;
     voltageMeter->unfiltered = state->voltageUnfiltered;
+#if defined(USE_BATTERY_VOLTAGE_SAG_COMPENSATION)
+    voltageMeter->sagFiltered = state->voltageSagFiltered;
+#endif
+}
+
+uint16_t getBatteryVoltageTaskFrequencyHz(void)
+{
+    uint16_t frequencyHz = SLOW_VOLTAGE_TASK_FREQ_HZ;
+#if defined(USE_BATTERY_VOLTAGE_SAG_COMPENSATION)
+    for (unsigned i = 0; i < PID_PROFILE_COUNT; i++) {
+        if (pidProfiles(i)->vbat_sag_compensation > 0) {
+            frequencyHz = FAST_VOLTAGE_TASK_FREQ_HZ;
+        }
+    }
```

Comment:
Instead of setting the high frequency for the task if **any** profile has the feature enabled. We could add the task reschedule in `mixerInitProfile()` and only check the current profile here. Then it will behave as expected and not run at a high frequency unless the active profile has the feature enabled. Will still need the reschedule call in `taskInit()` as it currently exists to catch the initial config since `mixerInit()` is called before the scheduler is initialized. But after that any profile changes would correctly set the task frequency.

Reply:
Not sure about this - I intentionally made this a 'global' setting, as to me it sounds like the lesser evil than the risk of having configurations with borderline system load, and getting spurious problems with some PID profiles but not with others. It will also affect how the display filtering works, based on the differing base frequency.
To me this is just a case where, if a pilot wants to use this feature, they have to pay the price of the higher system load caused for it.

Rescheduling tasks is the only way to do this currently anyway, as the tasks (defaults) list is stored in flash.

But happy to discuss.